src/lib.rs

```diff
@@ -1216,4 +1235,63 @@ mod tests {
             assert_eq!(output, "foo");
         }
     }
+
+    mod paths {
+        use super::*;
+        use pretty_assertions::assert_eq;
+        use std::path::Path;
+
+        fn write_test_script() -> PathBuf {
+            if cfg!(unix) {
+                let file = PathBuf::from("./test-script");
+                let script = "#!/usr/bin/env bash\necho test-output\n";
+                std::fs::write(&file, script).unwrap();
+                cmd_unit!(%"chmod +x test-script");
+                file
+            } else {
+                let file = PathBuf::from("./test-script.bat");
+                let script = "@echo test-output\n";
+                std::fs::write(&file, script).unwrap();
+                file
+            }
+        }
+
+        #[test]
+        fn ref_path_as_argument() {
+            in_temporary_directory(|| {
+                let file: &Path = Path::new("file");
+                std::fs::write(file, "test-contents").unwrap();
+                let StdoutUntrimmed(output) = cmd!("cat", file);
+                assert_eq!(output, "test-contents");
+            })
+        }
+
+        #[test]
+        fn ref_path_as_executable() {
+            in_temporary_directory(|| {
+                let file: &Path = &write_test_script();
+                let StdoutTrimmed(output) = cmd!(file);
```

Comment:
One weird detail with `Path` and `PathBuf` is this: If users create a path

- without slashes,
- that doesn't point to an existing file,
- they try to use that path as the executable in `cmd!`, and
- an executable with the same name can be found in one of the directories in `$PATH`,

then that executable from `$PATH` will be executed. This is a bit weird, since values of type `Path` or `PathBuf` are not supposed to point to files from `$PATH` that way. (As evidenced by `.exists()` returning `false`.) However `Command::new` behaves the same way, so I think this is weird, but ok.

Reply:
I think that's okay. I agree that it's conceptually weird, but you can also think of, for example `Path::new("echo")` as being a path to the `echo` binary, which is relative to one of the paths in `$PATH`.